network/src/main/scala/com/linkedin/norbert/network/client/ResponseHandlerComponent.scala

```diff
@@ -93,3 +95,8 @@ class ResponseProcessorMBeanImpl(clientName: Option[String], serviceName: String
   extends MBean(classOf[ResponseProcessorMBean], JMX.name(clientName, serviceName)) with ResponseProcessorMBean {
   def getQueueSize = queue.size
 }
+
+abstract class DarkCanaryResponseHandler()
+{
+  def upstreamCallback(isDark: Boolean, requestId: UUID, request: Request[Any, Any], message: NorbertProtos.NorbertMessage);
```

Comment:
s/message/response perhaps?
I think the request map somewhere within has the actual request. But I guess we should be able to mine that within the implementation.